examples/nlp/mlm_and_finetune_with_bert.py

```diff
@@ -0,0 +1,265 @@
+"""
+Title: End to End Masked Language Modeling & Fine-Tuning with BERT from Scratch
+Author: [Ankur Singh](https://twitter.com/ankur310794)
+Date created: 2020/09/03
+Last modified: 2020/17/03
+Description: Implement a Masked Language Modeling with BERT and fine-tune on IMDB Reviews dataset
+"""
+"""
+## Introduction
+Masked language modeling is a fill-in-the-blank task, where a model uses the context words surrounding a [MASK] token to try to predict what the [MASK] word should be. We will use IMDB Reviews raw text to pretrain and then use pretrained model weights to fine-tune sentiment classification.
+
+Steps:
+    1. Take raw text
+    2. Build pretraining BERT model from scratch using masked language model approach
+    3. Save pretrained model into .h5 file
+    4. Build classification model using BERT pretrained model (from step 3)
+    5. Evaluate classification model using end to end workflow
+"""
+
+"""
+## Setup
+Install tf-nightly via pip install tf-nightly.
+"""
+
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+from tensorflow.keras.layers.experimental.preprocessing import TextVectorization
+from dataclasses import dataclass
+import pandas as pd
+import numpy as np
+
+"""
+## Set-up Configuration
+"""
+
+
+@dataclass
+class Config:
+    MAX_LEN = 128
+    BATCH_SIZE = 16
+    LR = 1e-5
+    VOCAB_SIZE = 20000
+    EMBED_DIM = 32
+    NUM_HEAD = 2  # used in bert model
+    FF_DIM = 32  # used in bert model
+
+
+flags = Config()
+
+"""
+## Load Data
+"""
+
+"""shell
+wget https://github.com/LawrenceDuan/IMDb-Review-Analysis/raw/master/IMDb_Reviews.csv
+"""
+
+data = pd.read_csv("IMDb_Reviews.csv")
+
+"""
+## Dataset Preparation
```

Comment:
For each subsection, please add a text paragraph explaining what you're doing:

- How do you prepare the dataset? what do your samples end up looking like?
- What does the masked language model do (what are its inputs are outputs)?
- What does the classification model do?
- What does the end to end model do?
- How to interpret evaluation metrics?